apps/prairielearn/src/pages/instructorAssessmentManualGrading/assessmentQuestion/utils/useBatchActions.ts

```diff
@@ -0,0 +1,186 @@
+      console.error('Batch action failed:', error);
+      setErrorMessage(error instanceof Error ? error.message : 'An unknown error occurred');
+    },
+  });
+
+  // Handler for batch actions
+  const handleBatchAction = (actionData: BatchActionData, instanceQuestionIds: string[]) => {
+    if (instanceQuestionIds.length === 0) return;
+
+    batchActionMutation.mutate({
+      action: 'batch_action',
+      actionData,
+      instanceQuestionIds,
+    });
+  };
+
+  // Mutation for deleting all AI grading jobs
+  const deleteAiGradingJobsMutation = useMutation<
+    { success: boolean; numDeleted: number },
+    Error,
+    undefined
+  >({
+    mutationFn: async () => {
+      const response = await fetch('', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          __csrf_token: csrfToken,
+          __action: 'delete_ai_grading_jobs',
+        }),
+      });
+
+      if (!response.ok) {
+        throw new Error('Request failed with status ' + response.status);
+      }
+
+      const data = await response.json();
+      return data;
+    },
+    onSuccess: (data) => {
+      setErrorMessage(null);
+      setSuccessMessage(
+        `Deleted AI grading results for ${data.numDeleted} ${data.numDeleted === 1 ? 'question' : 'questions'}.`,
+      );
+      void queryClient.invalidateQueries({
+        queryKey: ['instance-questions', urlPrefix, assessmentId, assessmentQuestionId],
+      });
+    },
+    onError: (error) => {
+      console.error('Delete AI grading jobs failed:', error);
+      setErrorMessage(error instanceof Error ? error.message : 'An unknown error occurred');
+    },
+  });
+
+  // Mutation for deleting all AI instance question groupings
+  const deleteAiGroupingsMutation = useMutation<
+    { success: boolean; numDeleted: number },
+    Error,
+    undefined
+  >({
+    mutationFn: async () => {
+      const response = await fetch('', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          __csrf_token: csrfToken,
+          __action: 'delete_ai_instance_question_groupings',
+        }),
+      });
+
+      if (!response.ok) {
+        throw new Error('Request failed with status ' + response.status);
+      }
+
+      const data = await response.json();
+      return data;
+    },
+    onSuccess: (data) => {
+      setErrorMessage(null);
+      setSuccessMessage(
+        `Deleted AI submission grouping results for ${data.numDeleted} ${data.numDeleted === 1 ? 'question' : 'questions'}.`,
+      );
+      void queryClient.invalidateQueries({
+        queryKey: ['instance-questions', urlPrefix, assessmentId, assessmentQuestionId],
+      });
+    },
+    onError: (error) => {
+      console.error('Delete AI groupings failed:', error);
+      setErrorMessage(error instanceof Error ? error.message : 'An unknown error occurred');
```

Comment:
The experience here is weird. In the error case, the modal remains open, but the error is shown in an alert on the page _behind_ the modal?